src/test/java/com/eighthlight/EchoServerTest.java

```diff
@@ -2,10 +2,49 @@
 
 import org.junit.Test;
 import org.junit.Assert;
+import java.io.*;
 
 public class EchoServerTest {
     @Test
-    public void firstTest() {
-        Assert.assertEquals(1, 1);
+    public void createsSocketAndListensForClient() {
+        SocketWrapperSpy testSocketWrapper = new SocketWrapperSpy();
+        EchoServer testEchoServer = new EchoServer(testSocketWrapper);
+
+        testEchoServer.start(333);
+
+        Assert.assertTrue(testSocketWrapper.wasCreatedAndListened());
+    }
+
+    @Test
+    public void receivesDataFromClient() {
+        BufferedReader in = new BufferedReader(new StringReader("hello"));
+        SocketWrapperSpy testSocketWrapper = new SocketWrapperSpy(in);
+        EchoServer testEchoServer = new EchoServer(testSocketWrapper);
+
+        testEchoServer.start(333);
+
+        Assert.assertEquals("hello", testSocketWrapper.getReceivedData());
+    }
+
+    @Test
+    public void sendsReceivedDataBackToClient() {
+        BufferedReader in = new BufferedReader(new StringReader("hello"));
+        SocketWrapperSpy testSocketWrapper = new SocketWrapperSpy(in);
+        EchoServer testEchoServer = new EchoServer(testSocketWrapper);
+
+        testEchoServer.start(333);
+
+        Assert.assertEquals("hello", testSocketWrapper.getSentData());
+    }
+
+    @Test
+    public void stopsIfInputIsGoodbye(){
+        BufferedReader in = new BufferedReader(new StringReader("goodbye"));
```

Comment:
If you want to be really thorough on testing this, you can make some characters upper and lower case. Currently, if you removed the calls to `.toLowerCase().trim()`, the tests would still pass